### PR TITLE
Stack events

### PR DESCRIPTION
**Pull request by wlan0**
@alena1108

### CODE REVIEW

File: kubernetesevents/namespace_handler.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+	"github.com/rancher/go-rancher/client"
+	"github.com/rancher/kubernetes-agent/kubernetesclient"
+	"github.com/rancher/kubernetes-model/model"
+)
+
+const NamespaceKind string = "namespaces"
+const namespaceEventTypePrefix string = "stack."
+
+// Capable of handling RC and Service events
```
Comment (alena1108) [inline kubernetesevents/namespace_handler.go:17]: looks like this comment can be removed

File: kubernetesevents/namespace_handler.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+	return h.kindHandled
+}
+
+func (h *NamespaceHandler) Handle(event model.WatchEvent) error {
+	if i, ok := event.Object.(map[string]interface{}); ok {
+		var ns model.Namespace
+		mapstructure.Decode(i, &ns)
+		if ns == (model.Namespace{}) || ns.Spec == nil {
+			log.Errorf("Could not decode %+v to namespace", i)
+			return nil
```
Comment (alena1108) [inline kubernetesevents/namespace_handler.go:34]: do we return nil in situation like this, in other handlers?

Comment (wlan0) [inline kubernetesevents/namespace_handler.go:34]: yes, generic handler returns nil

File: kubernetesevents/generic_handler.go [hunk 3]
```diff
@@ -14,15 +14,7 @@ import (
...
-const eventTypePrefix string = "service."
-
-func NewHandler(rancherClient *client.RancherClient, kubernetesClient *kubernetesclient.Client, kindHandled string) *GenericHandler {
-	return &GenericHandler{
-		rancherClient: rancherClient,
-		kClient:       kubernetesClient,
-		kindHandled:   kindHandled,
-	}
-}
+const serviceEventTypePrefix string = "service."
```
Comment (ibuildthecloud) [inline kubernetesevents/generic_handler.go:17]: I don't understand why `generic_handler.go` was changed

File: kubernetesevents/namespace_handler.go [hunk 4]
```diff
@@ -0,0 +1,59 @@
+package kubernetesevents
```
Comment (ibuildthecloud) [inline kubernetesevents/namespace_handler.go:1]: Whats the fundamental difference between this and GenericHandler.  Why didn't GenericHandler suffice?

**Pull request closed without merging**